File: algorithms/algorithms-cplusplus/leetcode/Question_0415_Add_Strings.cc
#include <string>

using namespace std;

class Solution {
 public:
  string addStrings(string num1, string num2) {
    string res;
    int i = num1.length() - 1, j = num2.length() - 1, carry = 0;
    while (i >= 0 || j >= 0 || carry > 0) {
      carry += i >= 0 ? num1[i--] - '0' : 0;
      carry += j >= 0 ? num2[j--] - '0' : 0;
      res = to_string(carry % 10) + res;
      carry /= 10;
    }
    return res;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0415_Add_Strings.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0415_Add_Strings {
    public String addStrings(String num1, String num2) {
        if (num1 == null || num2 == null) {
            return null;
        }
        StringBuilder res = new StringBuilder();
        int i = num1.length() - 1, j = num2.length() - 1, carry = 0;
        while (i >= 0 || j >= 0 || carry > 0) {
            carry += i >= 0 ? num1.charAt(i--) - '0' : 0;
            carry += j >= 0 ? num2.charAt(j--) - '0' : 0;
            res.append(carry % 10);
            carry /= 10;
        }
        return res.reverse().toString();
    }
}
